### PR TITLE
float issue fixes for accumulate/distribute algo

### PR DESCRIPTION
**Pull request by avsek477**
This PR focuses on solving the float number calculation for order amounts involved in accumulate/distribute algorithm.

### CODE REVIEW

File: lib/accumulate_distribute/meta/init_state.js [hunk 1]
```diff
@@ -22,9 +23,9 @@ const initState = (args = {}) => {
   const orderAmounts = genOrderAmounts.gen(args)
 
   let totalAmount = 0
-  orderAmounts.forEach(a => { totalAmount += a })
+  orderAmounts.forEach(a => { totalAmount = nBN(totalAmount).plus(a).toNumber() })
```
Comment (robertkowalski) [inline lib/accumulate_distribute/meta/init_state.js:26]: test missing

File: lib/accumulate_distribute/meta/init_state.js [hunk 2]
```diff
@@ -22,9 +23,9 @@ const initState = (args = {}) => {
...
 
-  if (Math.abs(totalAmount - amount) > DUST) {
+  if (nBN(totalAmount).minus(amount).abs().toNumber() > DUST) {
```
Comment (robertkowalski) [inline lib/accumulate_distribute/meta/init_state.js:28]: abs() can introduce issues here:

lets say you have a remainder of 0.1
and you go to -0.1 next round
and dust is 0.05
then it does another round

File: lib/accumulate_distribute/util/gen_order_amounts.js [hunk 3]
```diff
@@ -24,17 +25,18 @@ const generateOrderAmounts = (args = {}) => {
   if (_isFinite(amountDistortion)) {
     let totalAmount = 0
 
-    while (Math.abs(amount - totalAmount) > DUST) {
+    while (nBN(amount).minus(totalAmount).abs().toNumber() > DUST) {
```
Comment (robertkowalski) [inline lib/accumulate_distribute/util/gen_order_amounts.js:28]: same here

File: lib/accumulate_distribute/util/gen_order_amounts.js [hunk 4]
```diff
@@ -24,17 +25,18 @@ const generateOrderAmounts = (args = {}) => {
...
       const m = Math.random() > 0.5 ? 1 : -1
-      const orderAmount = Math.min(sliceAmount, sliceAmount * (1 + (Math.random() * amountDistortion * m)))
-      const remAmount = amount - totalAmount
+      const randomDistortedPercentage = nBN(Math.random()).multipliedBy(amountDistortion).toNumber() * m + 1
```
Comment (robertkowalski) [inline lib/accumulate_distribute/util/gen_order_amounts.js:30]: is this a fix for something else? missing random dist?

**Review by vitaliystoliarovcc — APPROVED** [review]
lgtm

**Comment by avsek477** [comment]
merged in PR: https://github.com/bitfinexcom/bfx-hf-algo/pull/155

**Pull request closed without merging**